src/register_target.cpp

```diff
@@ -21,22 +21,42 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+#include <string>
 #include <unordered_map>
 #include <migraphx/register_target.hpp>
+#include <migraphx/ranges.hpp>
 
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 
+std::unordered_map<std::string, dynamic_loader>& target_lib_loader_map()
+{
+    static std::unordered_map<std::string, dynamic_loader> m; // NOLINT
```

Comment:
This can just be a `std::vector<dynamic_loader>`.

Reply:
It could be but we want to `dlclose()` when target library gets unloaded.  We can erase loader for that lib, and that would handle `dlcose()` properly.

Reply:
Ah yea, its probably faster to use a map to erase it.

Reply:
Ah wait, but you never erase from the map. Also, we cant remove when we unregister because it might not be in the map if the user links the lib or uses `dlopen`. Better to store as a vector and hide the global by making the function as `void store_target_lib(dynamic_loader lib)`.

Reply:
Done.